src/ssl/gadgets.cc

```diff
@@ -948,7 +943,6 @@ Ssl::CertificatesCmp(const Security::CertPointer &cert1, const Security::CertPoi
 const ASN1_BIT_STRING *
 Ssl::X509_get_signature(const Security::CertPointer &cert)
 {
-#if HAVE_LIBCRYPTO_X509_GET0_SIGNATURE
 #if SQUID_USE_CONST_X509_GET0_SIGNATURE_ARGS
```

Comment:
I gave this a lot of thought.  This function is already a wrapper to X509_get0_signature, whose sole purpose is to "[take] care of portability issues with older OpenSSL versions".  So, to avoid creating two wrappers, I've moved the function to compat/openssl.h, adapting the argument type to avoid needing to depend on other headers.  However, I think a simpler approach is just to use the `SQUID_CONST_X509_GET_SIGNATURE_ARGS` macro, either defined as `const` or empty, to handle the arguments type.  I'm leaving the commits for the individual options, so you can see what the code would be like.